### PR TITLE
Adjust permissions for deleting spaces

### PR DESCRIPTION
**Pull request by kobergj**
- Do not allow viewers and editors to delete a space
- Block deleting spaces via dav service (graph should be used)

### CODE REVIEW

**Review by micbar — APPROVED** [review]
LGTM

**Review by micbar — CHANGES_REQUESTED** [review]
I think the logic for the space root is not good enough.

File: internal/http/services/owncloud/ocdav/delete.go [hunk 1]
```diff
@@ -67,6 +67,15 @@ func (s *svc) handleDelete(ctx context.Context, w http.ResponseWriter, r *http.R
 
 	req := &provider.DeleteRequest{Ref: ref}
 
+	// do not allow deleting spaces via dav endpoint - use graph one instead
+	// we get a relative reference coming from the space root
+	// so if the path is "empty" we a referencing the space
+	if ref.Path == "." {
```
Comment (micbar) [inline internal/http/services/owncloud/ocdav/delete.go:73]: That could also be node further down in the tree. 

IMO a better check would be if storageID and opaqueID are the same.

Comment (kobergj) [reply internal/http/services/owncloud/ocdav/delete.go:73]: That is only working for decomposedfs. Eos for example is having different IDs for space and opaque. We need are more sophisticated way to determine a space root from other items.

For now it is fine though, the `ref` is created via `makeRelativeReference` which uses the space root and a path from there.

**Review by micbar — APPROVED** [review]

**Pull request merged**